ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/searchbar/clientlibs/js/searchbar.js

```diff
@@ -0,0 +1,247 @@
+        "https://searchautocompleteqa.magento-datasolutions.com/v0/LiveSearchAutocomplete.js"
+      );
+      // wait until script is loaded
+      await new Promise((resolve) => {
+        const interval = setInterval(() => {
+          if (window.LiveSearchAutocomplete) {
+            clearInterval(interval);
+            resolve();
+          }
+        }, 200);
+      });
+    }
+    const { dataServicesStorefrontInstanceContext } = this._state;
+    if (!dataServicesStorefrontInstanceContext) {
+      console.log("no dataServicesStorefrontInstanceContext");
+      return;
+    }
+    // initialize live-search
+    new window.LiveSearchAutocomplete({
+      environmentId: dataServicesStorefrontInstanceContext.environment_id,
+      websiteCode: dataServicesStorefrontInstanceContext.website_code,
+      storeCode: dataServicesStorefrontInstanceContext.store_code,
+      storeViewCode: dataServicesStorefrontInstanceContext.store_view_code,
+      config: {
+        pageSize: 8,
+        minQueryLength: "2",
+        currencySymbol: "$",
+        currencyRate: "1",
+        displayOutOfStock: true,
+        allowAllProducts: false,
+      },
+      context: {
+        customerGroup: dataServicesStorefrontInstanceContext.customer_group,
+      },
+    });
+
+    const formEle = document.getElementById("search_mini_form");
+
+    formEle.setAttribute(
+      "action",
+      `${dataServicesStorefrontInstanceContext.store_url}catalogsearch/result`
+    );
+    // initialize store event after live-search
+    this._initStoreEvent();
+  }
+
+  async _initStoreEvent() {
+    //  Magento Store event
+    if (!window.magentoStorefrontEvents) {
+      this._injectStoreScript(
+        "https://unpkg.com/@adobe/magento-storefront-events-sdk@qa/dist/index.js"
+      );
+      this._injectStoreScript(
+        "https://unpkg.com/@adobe/magento-storefront-event-collector@qa/dist/index.js"
+      );
+    }
+    // wait until script is loaded
+    await new Promise((resolve) => {
+      const interval = setInterval(() => {
+        if (window.magentoStorefrontEvents) {
+          clearInterval(interval);
+          resolve();
+        }
+      }, 200);
+    });
+
+    const mse = window.magentoStorefrontEvents;
+    const { dataServicesStorefrontInstanceContext, storeConfig } = this._state;
+
+    const {
+      catalog_extension_version,
+      environment,
+      environment_id,
+      store_code, // TODO: storeCode is also in storeConfig w/ diff value
+      store_id,
+      store_name,
+      store_url,
+      store_view_code,
+      store_view_id,
+      store_view_name,
+      website_code,
+      website_id,
+      website_name,
+    } = dataServicesStorefrontInstanceContext;
+    const { baseCurrencyCode /* , storeCode */ } = storeConfig;
+
+    mse.context.setMagentoExtension({
+      magentoExtensionVersion: this._state.magentoExtensionVersion,
+    });
+    // mse.context.setShopper({ shopperId: "logged-in" }); // TODO:
```

Comment:
If we would use the storefront events sdk from the react part, this would already be implemented: https://github.com/adobe/aem-core-cif-components/blob/330af014c93863499de53eff3431ba047248259c/extensions/experience-platform-connector/src/events-collector/EventCollectorContext.js#L82-L101

Reply:
got it, so I guess it make sense just to remove whole `async _initStoreEvent(` block then? since the event also gets set here then this function isnt doing anything.